src/qtbase/QColor.js

```diff
@@ -1,32 +1,78 @@
-// TODO complete implementation (with attributes `r`,`g` and `b`).
-
 class QColor {
   constructor(val) {
+    this.r = this.g = this.b = 0;
+    this.a = 1;
     this.$value = "black";
     if (val instanceof QColor) {
       // Copy constructor
-      this.$value = val.$value;
+      this.a = val.a;
+      this.r = val.r;
+      this.g = val.g;
+      this.b = val.b;
     } else if (typeof val === "string") {
-      this.$value = val.toLowerCase();
+      const lval = val.toLowerCase();
+      if (QColor.colormap[lval]) {
+        const rgb = QColor.colormap[lval];
+        this.r = rgb[0] / 255;
+        this.g = rgb[1] / 255;
+        this.b = rgb[2] / 255;
+      } else if (lval === "transparent") {
+        this.a = 0;
+      } else if (lval[0] === "#") {
+        const hex = lval.substr(1);
+        if (hex.length === 3) {
+          this.r = parseInt(hex[0], 16) / 15;
+          this.g = parseInt(hex[1], 16) / 15;
+          this.b = parseInt(hex[2], 16) / 15;
+        } else {
+          const rgb = hex.match(/.{2}/g).map(x => parseInt(x, 16));
+          if (rgb.length === 4) {
+            this.a = rgb.shift() / 255;
+          }
+          this.r = rgb[0] / 255;
+          this.g = rgb[1] / 255;
+          this.b = rgb[2] / 255;
+        }
+      } else {
+        throw new Error(`Can not convert ${val} to color`);
+      }
     } else if (typeof val === "number") {
       // we assume it is int value and must be converted to css hex with padding
-      const rgb = (Math.round(val) + 0x1000000).toString(16).substr(-6);
-      this.$value = `#${rgb}`;
+      const hex = (Math.round(val) + 0x1000000).toString(16).substr(-6);
+      const rgb = hex.match(/.{2}/g).map(x => parseInt(x, 16));
+      this.r = rgb[0] / 255;
+      this.g = rgb[1] / 255;
+      this.b = rgb[2] / 255;
     }
   }
   toString() {
```

Comment:
you could cache result. 
Especially the $ css function. can bring significant performance improvements in the animation.